src/main/java/com/zaxxer/hikari/pool/PoolEntry.java

```diff
@@ -90,7 +90,9 @@ void setFutureEol(final ScheduledFuture<?> endOfLife)
 
    Connection createProxyConnection(final ProxyLeakTask leakTask, final long now)
    {
-      return ProxyFactory.getProxyConnection(this, connection, openStatements, leakTask, now, hikariPool.isAutoCommit());
+      ProxyConnection pc = ProxyFactory.getProxyConnection(this, connection, openStatements, leakTask, now);
+      hikariPool.initProxyConnection(pc);
```

Comment:
EDIT: but with two members being initialized, maybe it comes out even.  I will have to benchmark before this merge.